src/structures/GuildBan.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const Base = require('./Base');
+
+/**
+ * Represents a ban in a guild on Discord.
+ * @extends {Base}
+ */
+class GuildBan extends Base {
+  /**
+   * @param {Client} client The instantiating client
+   * @param {Object} data The data for the ban
+   * @param {Guild} guild The guild in which the ban is
+   */
+  constructor(client, data, guild) {
+    super(client);
+
+    /**
+     * The guild in which the ban is
+     * @type {Guild}
+     */
+    this.guild = guild;
+
+    /**
+     * The reason for the ban
+     * @type {?string}
+     */
+    this.reason = null;
+
+    /**
+     * The user this ban applies to
+     * @type {User}
+     */
+    this.user = null;
+
+    if (data) this._patch(data);
+  }
+
+  _patch(data) {
+    if ('user' in data) this.user = this.client.users.add(data.user, true);
+    if ('reason' in data) this.reason = data.reason;
+  }
+
+  /**
+   * Whether this GuildBan is a partial
+   * If the reason is not provided the value is null
+   * @type {boolean}
+   * @readonly
+   */
+  get partial() {
+    return typeof this.reason === 'undefined';
+  }
```

Comment:
```suggestion
  /**
   * Whether this GuildBan is a partial
   * If the reason is not provided the value is null
   * @type {boolean}
   * @readonly
   */
  get partial() {
    return this.reason === null;
  }
```

Reply:
this is incorrect, if the reason is `null` then it wasn't set by the person who created the ban (aka the object is *not* partial), only case in which a GuildBan is partial is when the reason is `undefined`